pallets/sponsorships/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+		OptionQuery,
+	>;
+
+	/// Pallets use events to inform users when important changes are made.
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// An account has been sponsored.
+		NewSponsorship {
+			/// The sponsor account.
+			sponsor: AccountIdOf<T>,
+			/// The sponsored account.
+			beneficiary: AccountIdOf<T>,
+			/// The sponsored amount
+			amount: BalanceOf<T>,
+		},
+		/// An account is no longer sponsored.
+		SponsorshipRemoved {
+			/// The account no longer being the sponsor.
+			was_sponsor: AccountIdOf<T>,
+			/// The account no longer being the sponsored.
+			was_beneficiary: AccountIdOf<T>,
+		},
+		/// An existing sponsorship has been modified.
+		SponsorshipUpdated {
+			/// The sponsor account.
+			sponsor: AccountIdOf<T>,
+			/// The sponsored account.
+			beneficiary: AccountIdOf<T>,
+			/// The previous sponsored amount.
+			old_amount: BalanceOf<T>,
+			/// The updated sponsored amount.
+			new_amount: BalanceOf<T>,
+		},
+		/// A call has been sponsored.
+		CallSponsored {
+			/// The account paying for the fee costs.
+			by: AccountIdOf<T>,
+		},
+	}
+
+	/// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The account is already being sponsored.
+		AlreadySponsored,
+		/// This action cannot be sponsored.
+		CantSponsor,
+		/// The sponsorship doesn't exist.
+		UnknownSponsorship,
+		/// The cost is higher than the max sponsored.
+		SponsorshipOutOfLimits,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a new sponsorship for `beneficiary` with a value of `amount`.
+		///
+		/// Parameters
+		/// - `beneficiary`: Account to be sponsored by the caller.
+		/// - `amount`: How much `beneficiary` is sponsored for.
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::sponsor_account())]
+		pub fn sponsor_account(
+			origin: OriginFor<T>,
+			beneficiary: AccountIdOf<T>,
+			amount: BalanceOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			match <Sponsorships<T>>::get(&who, &beneficiary) {
+				Some(_) => return Err(Error::<T>::AlreadySponsored.into()),
+				None => {},
+			}
+			// TODO: Reserve SponsorshipDeposit
+			// Register new sponsorship.
+			<Sponsorships<T>>::set(&who, &beneficiary, Some(amount));
+			frame_system::Pallet::<T>::inc_providers(&beneficiary);
+			Self::deposit_event(Event::NewSponsorship { sponsor: who, beneficiary, amount });
+			Ok(())
+		}
+
+		/// Remove an account from the list of sponsored accounts managed by origin.
+		///
+		/// Parameters
+		/// - `beneficiary`: Account to be removed from a sponsorship.
+		#[pallet::call_index(1)]
+		#[pallet::weight(<T as Config>::WeightInfo::remove_sponsorship())]
+		pub fn remove_sponsorship_for(
+			origin: OriginFor<T>,
+			beneficiary: AccountIdOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			if <Sponsorships<T>>::take(&who, &beneficiary).is_some() {
```

Comment:
Original comment at: https://github.com/r0gue-io/pop-node/pull/351#discussion_r1816353978

What would happen to the weight here? Is it just lost?